python/doc_generator.py

```diff
@@ -47,13 +76,51 @@ def _get_transform_markdown(transform_yaml: Dict,
         md.h2('Parameters'),
         md.table(transform_yaml['arguments']),
         md.h2('Example'),
+        md.text(DATASET_PREVIEW_TEXT) if 'preview-data' in transform_yaml else None,
+        md.python_code(DATASET_PREVIEW_CODE) if 'preview-data' in transform_yaml else None,
+        t1,# md.text(transform_yaml['preview-data']) if 'preview-data' in transform_yaml else None,
+        md.text(TRANSFORMED_DATA_PREVIEW_TEXT) if 'preview-data' in transform_yaml else None,
         md.python_code(transform_yaml['example_code']),
+        t2,# md.text(transform_yaml['post-transform-data']) if 'preview-data' in transform_yaml else None,
         md.h2('Source Code'),
         md.github_url(transform_type_dir_name, transform_name),
         '',
     ]
-    return '\n\n'.join(markdown_elements)
+    return '\n\n'.join(filter(lambda x: x!=None, markdown_elements))
 
+def _generate_dataset_markdowns(
+    rasgo: Any,
+    dataset_id: int,
+    output_cols: List[str],
+    transform_details: Dict
+) -> List[str]:
+    """Creates markdown representation of both a pre-transform Rasgo Dataset
+    and a transformed Rasgo Dataset
+
+    Arguments:
+        :dataset_id: int: A Rasgo Community Dataset ID matching the columns named 
+        in the `example_code` section of the YAML doc
+        :output_cols: List[str]: A list of column names to include in the output
+        YAML representation
+        :transform_args: dict: TODO what do we want here? How to build the transform from args
+    """
+    ds = rasgo.get.dataset(dataset_id)
+    ds_md = ds.preview()[output_cols].to_markdown()
+
+    ds1 = ds.transform(
+        transform_name=transform_details["transform-name"],
+        arguments=transform_details["transform-args"]
+    )
+    ds1_md = ds1.preview().to_markdown()
+
+    return[ds_md, ds1_md]
```

Comment:
Since we're returning two strings every time let's change the return type to `Tuple[str, str]` instead of a list of strings of any size.

Need to change `[ds_md, ds1_md]` to `ds_md, ds1_md` as well

Reply:
I wanted a list but I actually think you're right - structured and ordered would make more sense for this return data type than unstructured and ordered